### PR TITLE
Use LZMA instead of zlib

### PR DESCRIPTION
**Pull request by rbarry82**
LZMA is about 10% more efficient in the worst case, and only increases as complexity goes up

### CODE REVIEW

File: tests/test_dashboard_provider.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 RELATION_TEMPLATES_DATA = {
     "file:first": {
         "charm": "provider-tester",
-        "content": "eNorSS0uiU/LLCou4QIAG8EEUg==",
+        "content": "/Td6WFoAAATm1rRGAgAhARYAAAB0L+WjAQAKdGVzdF9maXJzdAoAAIC4BxCQe2GHAAEjC8Ib/QkftvN9AQAAAAAEWVo=",
```
Comment (jameinel) [inline tests/test_dashboard_provider.py:22]: Interesting that we did this to 'make it more efficient' but in the case of short strings, it is clearly much much less :)

**Comment by jameinel** [comment]
(forget my earlier XZ statement. I do believe there was an issue around it,
but it seems my memory for what is recommended was incorrect. Also Python
uses lzma to do .xz decoding.)

File: tests/test_dashboard_provider.py (hunk 1 above)
Comment (Abuelodelanada) [reply tests/test_dashboard_provider.py:22]: This is true, but we are compressing grafana dashboards that are kilometric json objects. So if I understand correctly most of the time we are going to see the profit.

**Review by Abuelodelanada — APPROVED** [review]
Looks good to me

**Review by Abuelodelanada — APPROVED** [review]
LGTM

**Review by simskij — APPROVED** [review]

**Pull request merged**